gobblin-core/build.gradle

```diff
@@ -51,7 +51,7 @@ dependencies {
 
   if (project.hasProperty('useHadoop2')) {
     compile externalDependency.avroMapredH2
-    testRuntime externalDependency.hadoopAws
+    compile externalDependency.hadoopAws
```

Comment:
There isn't an explicit reference, but as soon as you use a uri like `s3a://...` for one of the fs.uri properties (in my case `data.publisher.fs.uri`) there will be a runtime error because the hadoop-aws library is missing.